linkerd/app/integration/src/tests/discovery.rs

```diff
@@ -422,66 +426,71 @@ mod http2 {
-            .run()
-            .await;
-        let srv1_addr = srv1.addr;
+        let svc_addr = SocketAddr::from(([127, 0, 0, 1], 8080));
+        let alpha = server::http2().route("/", "alpha").run().await;
+        let beta = server::http2().route("/", "beta").run().await;
 
         // Start with the first server.
         let dstctl = controller::new();
         let polctl = controller::policy();
-        let (_profile, _policy, dst) = send_default_dst(&dstctl, &polctl, &srv1);
+        let (_profile, _policy, dst) = send_default_dst(&dstctl, &polctl, &alpha, Some(svc_addr));
 
         let proxy = proxy::new()
-            .outbound_ip(srv1.addr)
+            .outbound_ip(svc_addr)
             .controller(dstctl.run().await)
             .policy(polctl.run().await)
             .run()
             .await;
         let client = client::http2(proxy.outbound, HOST);
         let metrics = client::http1(proxy.admin, "localhost");
 
-        assert_eq!(client.get("/").await, "hello");
+        assert_eq!(client.get("/").await, "alpha");
 
         // Simulate the first server falling over without discovery
         // knowing about it...
-        srv1.join().await;
-        tokio::task::yield_now().await;
+        tracing::info!(%alpha.addr, "Stopping");
+        let alpha_addr = alpha.addr;
+        alpha.join().await;
 
-        // Wait until the proxy has seen the `srv1` disconnect...
+        // Wait until the proxy has seen the `alpha` disconnect...
         metrics::metric("tcp_close_total")
             .label("peer", "dst")
             .label("direction", "outbound")
-            .label("target_addr", srv1_addr.to_string())
+            .label("target_addr", alpha_addr.to_string())
             .value(1u64)
             .assert_in(&metrics)
             .await;
+        tracing::info!("Connection closed");
 
         // Start a new request to the destination, now that the server is dead.
         // This request should be waiting at the balancer for a ready endpoint.
         //
         // The only one it knows about is dead, so it won't have progressed.
-        let fut = client.request(client.request_builder("/bye"));
+        tracing::info!("Sending request");
+        let fut = client.request(client.request_builder("/"));
 
         // When we tell the balancer about a new endpoint, it should have added
         // it and then dispatched the request...
-        dst.send_addr(srv2.addr);
+        tracing::info!(%beta.addr, "Adding");
+        dst.send_addr(beta.addr);
 
-        let res = fut.await.expect("/bye response");
+        let res = fut.await.expect("beta response");
         assert_eq!(res.status(), http::StatusCode::OK);
+        assert_eq!(
+            String::from_utf8(
+                hyper::body::to_bytes(res.into_body())
+                    .await
+                    .unwrap()
+                    .to_vec(),
+            )
+            .unwrap(),
+            "beta"
+        );
```

Comment:
tiny nit: there's [a `body_string` function in the test utils](https://github.com/linkerd/linkerd2-proxy/blob/8ce2df4faea100cc53db1a8f6f0822555fb9c685/linkerd/app/test/src/http_util.rs#L115-L127) that we could just use here: 
```suggestion
        assert_eq!(http_util::body_string(res.into_body()).await, "beta");
```